### PR TITLE
GH-81: Add TCP Sink

### LINKED ISSUE #81: Port TCP as s-c-s sink
As a developer, I'd like to move *tcp* module from XD to s-c-s repo, so I can use it as sink to build streaming pipeline. 
**Acceptance:**
* *tcp* as sink module is available in *spring-cloud-stream-modules* repo 
* maven repo includes *tcp* module with respective group/artifact/version detail

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves spring-cloud/spring-cloud-stream-modules#81 

**The common code is identical to that in** #130

### CODE REVIEW

**Comment by artembilan** [comment]
I'd like merge #130 before review this one.

**Comment by sabbyanandan** [comment]
Thanks, @artembilan! :smile:

**Comment by garyrussell** [comment]
Rebased; pushed changes from #130

**Comment by garyrussell** [comment]
Repushed polish - missed a file.

**Comment by artembilan** [comment]
Please, rebase one more time.

**Comment by garyrussell** [comment]
Rebased

**Comment by garyrussell** [comment]
Pushed docs and a missing property on the source.

File: spring-cloud-stream-modules-docs/src/main/asciidoc/sources.adoc [hunk 1]
```diff
@@ -145,6 +145,42 @@ $$tmpFileSuffix$$:: $$extension to use when downloading files$$ *($$String$$, de
 $$user$$:: $$the username to use$$ *($$String$$, no default)*
 $$withMarkers$$:: $$if true emits start of file/end of file marker messages before/after the data. Only valid with FileReadingMode 'lines'$$ *($$Boolean$$, no default)*
 
+[[spring-cloud-stream-modules-source-tcp]]
+=== TCP
+The `tcp` source acts as a server and allows a remote party to connect to XD and submit data over a raw tcp socket.
```
Comment (artembilan) [inline spring-cloud-stream-modules-docs/src/main/asciidoc/sources.adoc:150]: `XD` is wrong word in this context.
Sorry, I don't have any objections and antisympathy to XD, but we don't talk in this sentence about XD at all.

File: tcp-sink/README.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+== Build
+
+```
+$> mvn package
+```
+
+== Run
+
+```
+$> java -jar target/tcp-source-${version}-exec.jar
```
Comment (artembilan) [inline tcp-sink/README.adoc:23]: `tcp-sink` I guess.
What is the purpose of `exec` suffix ?

**Comment by artembilan** [comment]
That's all. 
I can fix all those my concerns on merge:

```
[INFO] --- animal-sniffer-maven-plugin:1.14:check (enforce-java-7) @ tcp-sink ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 25.678s
```

I found the 

```
<configuration>
    <classifier>exec</classifier>
</configuration>
```

In the `spring-cloud-stream-modules` parent POM for the `spring-boot-maven-plugin`.

File: spring-cloud-stream-modules-docs/src/main/asciidoc/sources.adoc (hunk 1 above)
Comment (garyrussell) [inline spring-cloud-stream-modules-docs/src/main/asciidoc/sources.adoc:150]: oops

**Comment by garyrussell** [comment]
> I can fix 

Thanks

**Comment by artembilan** [comment]
Merged as https://github.com/spring-cloud/spring-cloud-stream-modules/commit/054e1e537ac229b3c0cb675a953a36f422034326

**Pull request closed without merging**